src/main/java/com/jcabi/maven/plugin/UnwovenClasses.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2012-2015, jcabi.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.jcabi.maven.plugin;
+
+import com.jcabi.log.Logger;
+import java.io.File;
+import java.io.IOException;
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.plugin.MojoFailureException;
+
+/**
+ * Operations on the unwoven classes.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ *
+ */
+public final class UnwovenClasses {
+    /**
+     * Directory where unwoven classes are saved.
+     */
+    private final transient File unwoven;
+
+    /**
+     * Output directory from where the classes are taken.
+     */
+    private final transient File classes;
+
+    /**
+     * Maven execution phase.
+     */
+    private final transient String phase;
+
+    /**
+     * Constructor.
+     * @param uvn Dir where unwoven classes go
+     * @param cls Directory where the classes are found
+     * @param mph Maven execution phase
+     */
+    public UnwovenClasses(final File uvn, final File cls,
+        final String mph) {
+        this.unwoven = uvn;
+        this.classes = cls;
+        this.phase = mph;
+    }
+
+    /**
+     * Perform the copy.
```

Comment:
@amihaiemil I think we should describe more here why we do the copying twice to different directories